WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java

```diff
@@ -262,19 +262,24 @@ public static void addOrUpdatePost(ReaderPost post) {
     }
 
     public static ReaderPost getBlogPost(long blogId, long postId, boolean excludeTextColumn) {
-        return getPost(false, blogId, postId, excludeTextColumn);
+        return getPost("blog_id=? AND post_id=?", new String[]{Long.toString(blogId), Long.toString(postId)},
+                excludeTextColumn);
+    }
+
+    public static ReaderPost getBlogPost(String blogSlug, String postSlug, boolean excludeTextColumn) {
+        return getPost("blog_url LIKE ? AND url LIKE ?", new String[] {"%//" + blogSlug, "%/" + postSlug + "/"},
```

Comment:
I asked around and it seems that the URL cannot be guaranteed to include the trailing slash. Let's open this as an issue to address after the PR is merged, wdyt?